modules/core/src/main/java/org/apache/ignite/internal/processors/cache/checker/processor/PartitionReconciliationProcessor.java

```diff
@@ -0,0 +1,600 @@
+                }
+
+                PipelineWorkload workload = takeTask();
+
+                workProgress.printWorkProgress();
+
+                if (workload instanceof Batch)
+                    handle((Batch)workload);
+                else if (workload instanceof Recheck)
+                    handle((Recheck)workload);
+                else if (workload instanceof Repair)
+                    handle((Repair)workload);
+                else
+                    log.error("Unsupported workload type: " + workload);
+            }
+
+            return new ExecutionResult<>(prepareResult());
+        }
+        catch (InterruptedException | IgniteException e) {
+            String errMsg = "Partition reconciliation was interrupted.";
+
+            waitWorkFinish();
+
+            log.warning(errMsg, e);
+
+            return new ExecutionResult<>(prepareResult(), errMsg + " " + String.format(ERROR_REASON, e.getMessage(), e.getClass()));
+        }
+        catch (Exception e) {
+            String errMsg = "Unexpected error.";
+
+            log.error(errMsg, e);
+
+            return new ExecutionResult<>(prepareResult(), errMsg + " " + String.format(ERROR_REASON, e.getMessage(), e.getClass()));
+        }
+    }
+
+    /**
+     * @param workload Workload.
+     */
+    private void handle(Batch workload) throws InterruptedException {
+        compute(
+            CollectPartitionKeysByBatchTask.class,
+            new PartitionBatchRequest(workload.sessionId(), workload.workloadChainId(), workload.cacheName(), workload.partitionId(), batchSize, workload.lowerKey(), startTopVer),
+            res -> {
+                KeyCacheObject nextBatchKey = res.get1();
+
+                Map<KeyCacheObject, Map<UUID, GridCacheVersion>> recheckKeys = res.get2();
+
+                assert nextBatchKey != null || recheckKeys.isEmpty();
+
+                if (nextBatchKey != null)
+                    schedule(new Batch(workload.sessionId(), workload.workloadChainId(), workload.cacheName(), workload.partitionId(), nextBatchKey));
+
+                if (!recheckKeys.isEmpty())
+                    schedule(
+                        new Recheck(workload.sessionId(), workload.workloadChainId(), recheckKeys,
+                            workload.cacheName(), workload.partitionId(), 0, 0),
+                        recheckDelay,
+                        TimeUnit.SECONDS
+                    );
+            }
+        );
+    }
+
+    /**
+     * @param workload Workload.
+     */
+    private void handle(Recheck workload) throws InterruptedException {
+        compute(
+            CollectPartitionKeysByRecheckRequestTask.class,
+            new RecheckRequest(workload.sessionId(), workload.workloadChainId(), new ArrayList<>(workload.recheckKeys().keySet()), workload.cacheName(),
+                workload.partitionId(), startTopVer),
+            actualKeys -> {
+                Map<KeyCacheObject, Map<UUID, GridCacheVersion>> notResolvingConflicts
```

Comment:
May be extract body of this function to the method in order to improve readability?